src/middlewared/middlewared/event.py

```diff
@@ -2,18 +2,20 @@
 import contextlib
 import json
 import threading
+import typing
 
+from middlewared.role import RoleManager
 from middlewared.schema import Any, clean_and_validate_arg, ValidationErrors
 from middlewared.settings import conf
 
 
 class Events:
-    def __init__(self, role_manager):
+    def __init__(self, role_manager: RoleManager):
         self.role_manager = role_manager
-        self._events = {}
-        self.__events_private = set()
+        self._events: typing.Dict[str, dict[str, typing.Any]] = {}
+        self.__events_private: typing.Set[str] = set()
 
-    def register(self, name, description, private, returns, no_auth_required, no_authz_required, roles):
+    def register(self, name: str, description: str, private: bool, returns, no_auth_required, no_authz_required, roles: typing.Iterable[str]):
```

Comment:
`no_auth_required/no_authz_required` have been left out ?

Reply:
If I came across anything that I wasn't sure about, I just skipped it. Figured it's better to be less specific than wrong, plus I plan on spending the whole week working on type hints/docstrings so I can go back to it.